lib/cursor.js

```diff
@@ -914,38 +912,18 @@ class Cursor extends CoreCursor {
     if (typeof options === 'function') (callback = options), (options = {});
     options = Object.assign({}, { skipKillCursors: false }, options);
 
-    this.s.state = CursorState.CLOSED;
-    if (!options.skipKillCursors) {
-      // Kill the cursor
-      this.kill();
-    }
-
-    const completeClose = () => {
-      // Emit the close event for the cursor
-      this.emit('close');
-
-      // Callback if provided
-      if (typeof callback === 'function') {
-        return handleCallback(callback, null, this);
-      }
-
-      // Return a Promise
-      return new this.s.promiseLibrary(resolve => {
-        resolve();
-      });
-    };
-
-    if (this.cursorState.session) {
-      if (typeof callback === 'function') {
-        return this._endSession(() => completeClose());
+    return maybePromise(this, callback, cb => {
+      this.s.state = CursorState.CLOSED;
+      if (!options.skipKillCursors) {
+        // Kill the cursor
+        this.kill();
       }
 
-      return new this.s.promiseLibrary(resolve => {
-        this._endSession(() => completeClose().then(resolve));
+      this._endSession(() => {
+        this.emit('close');
```

Comment:
I'm not 100% how session works, but just wanted to bring this to your attention. Is it possible we might need a `this.emit('close');`  outside of the `this._endSession` if there is no session?

B/C the previous code had an emit in the `completeClose` https://github.com/mongodb/node-mongodb-native/blob/3.5/lib/cursor.js#L923-L948

Reply:
[_endSession](https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/core/cursor.js#L373-L398) always calls its callback, even if its not actually sending a command to the server. I had originally maintained the two separate paths that the old code took, but I think the result is the same. The difference now is that you'll always see `_endSession` in the stacktrace, but I imagine that was the case most of the time previously.

Reply:
Awesome, thanks for the explanation.